ha/setup/create_pacemaker_resources.py

```diff
@@ -302,14 +316,71 @@ def uds(cib_xml, push=False, **kwargs):
-            cib_push(cib_xml)
+def config_constraint(cib_xml, push=False, **kwargs):
+    """
+    Configure all constaints
+
+    Args:
+        cib_xml (str): cib cluster file.
+    """
+    constraints = [
+        f"pcs -f {cib_xml} constraint order io_group-clone then motr-free-space-mon",
+        f"pcs -f {cib_xml} constraint colocation add motr-free-space-mon with io_group-clone",
+        f"pcs -f {cib_xml} constraint colocation add s3backprod with io_group-clone"
+        # According to EOS-9258, there is a bug which requires UDS to be started after csm_agent
+        f"pcs -f {cib_xml} constraint colocation add uds with csm-agent score=INFINITY",
+        f"pcs -f {cib_xml} constraint order csm-agent then uds"
+    ]
+    with_uds = kwargs["uds"] if "uds" in kwargs else False
+    for c in constraints:
+        if with_uds == False and "uds" in c:
+            continue
+        process.run_cmd(c)
+    if push:
+        cib_push(cib_xml)
 
-core_io = [hax, motr_conf, motr, s3auth, s3servers, haproxy]
+def configure_stonith(cib_xml=None, push=False, **kwargs):
+    """Create ipmi stonith resource."""
+    try:
+        if cib_xml is None:
+            cib_xml = cib_get(const.CIB_FILE)
+
+        stonith_config = kwargs.get("stonith_config")
+        if stonith_config:
+            Log.info("Configuring stonith.")
+            resource_id = stonith_config.get("resource_id")
+            node_name = stonith_config.get("node_name")
+
+            # check for stonith config present for that node
+            _output, _err, _rc = process.run_cmd(f"pcs stonith show {resource_id}", check_error=False)
+            if _output and "Error" not in _output and resource_id in _output:
+                Log.info(f"Stonith configuration already exists for node {node_name}.")
+                return
+
+            ipaddr = stonith_config.get("ipaddr")
+            login = stonith_config.get("login")
+            passwd = stonith_config.get("passwd")
+            pcmk_host_list = stonith_config.get("pcmk_host_list")
+            auth = stonith_config.get("auth")
+
+            process.run_cmd(f"pcs -f {cib_xml} stonith create {resource_id} fence_ipmilan ipaddr={ipaddr} \
+                login={login} passwd={passwd} pcmk_host_list={pcmk_host_list} pcmk_host_check=static-list \
+                lanplus=true auth={auth} power_timeout=40 verbose=true \
+                op monitor interval=10s meta failure-timeout=15s --group ha_group")
+            process.run_cmd(f"pcs -f {cib_xml} constraint location {resource_id} avoids {pcmk_host_list}")
+            process.run_cmd(f"pcs -f {cib_xml} resource clone {resource_id}")
+
+            if push:
+                cib_push(cib_xml)
+        else:
+            Log.info("Stonith configuration not available to create stonith resource.")
+
+    except Exception as e:
+        raise ConfigureStonithResourceError(f"Failed to create stonith resource, Error: {e}")
+
+
+core_io = [hax, motr, s3auth, s3servers, haproxy]
```

Comment:
Why "motr_conf" is deleted here? Is it part of this task?

Reply:
change was came through rebase, so updated it.